app/controllers/application_controller.rb

```diff
@@ -2,4 +2,18 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  rescue_from Exception, with: :handle_exception
+
+  before_filter :add_allow_credentials_headers
+
+
+  def handle_exception(exception)
```

Comment:
in my opinion method name not reflects inside actions.
this method not handling exception, it's just redirecting when something happens, so it's should be renamed to something like this: redirect_on_exception
btw, argument should be used somehow, otherwise it should not be transferred here.